docs(toggle-group): Adds example documentation. (#10309)
patternfly/patternfly-react: packages/react-core/src/components/ToggleGroup/examples/ToggleGroup.md

## Examples

### Single select toggle group

A single select toggle group allows users to toggle between multiple items.

> To support single selection, use the `onChange` callback to pass the selected `<ToggleGroupItem>` to `isSelected`.

Reviewer: For this and the multi select example, maybe the key point should be more that `isSelected` needs to be true. The onChange is sort of facilitating that in these examples, but an item could be selected just by passing `isSelected={true}` or use something other than the onChange to update an item's selected state.

Maybe something along the lines of:

```suggestion
To indicate whether a `<ToggleGroupItem>` is selected or not, use the `isSelected` property.
```

Author: ah, ty this is very helpful! Your suggestion sounds great

```ts file="./ToggleGroupDefaultSingle.tsx"
```

### Multi select toggle group

A multi select toggle group allows users to select multiple items at once.

When a toggle item is disabled it cannot be selected. Click the "Disable all" button in the following example to see this in action.

> To support multiple selection, use the `onChange` callback to pass each selected `<ToggleGroupItem>` to `isSelected`.

Reviewer: Then related to the above comment, maybe this line could be removed?

```ts file="./ToggleGroupDefaultMultiple.tsx"
```

### With icons

You can use a recognizable icon as a toggle item label.

To do this, pass an imported icon to the `icon` property of a `<ToggleGroupItem>`.

```ts file="./ToggleGroupIcon.tsx"
```

### With text and icons

Adding text to a toggle item with an icon helps clarify the button's function.

To do this, pass a descriptive label to the `text` property of a `<ToggleGroupItem>`.

> When passing both `text` and `icon` properties to a `<ToggleGroupItem>`, you can also pass in `iconPosition` to determine whether the icon is rendered at the start or end of the item.

Reviewer: Since the update that added this prop was in the v6 branch, this line should be removed for this PR, but we would want to add it to the v6 branch whenever this update gets pulled into v6 (or we could open a separate PR that adds this content to v6).

Author: Ah, that makes sense. Do you think I can just link that additional PR to the same issue, or is a new issue needed?
